src/components/button/Button.tsx

```
@@ -0,0 +1,114 @@
+    backgroundColor: interactColor
+  },
+  ':active': {
+    backgroundColor: baseColor
+  }
+})
+
+const StyledButton = styled('button', {
+  background: 'unset',
+  border: 'none',
+  borderRadius: '$1',
+  cursor: 'pointer',
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  fontFamily: 'sans',
+  fontSize: 'md',
+  fontWeight: 500,
+  height: '$2',
+  letterSpacing: '0.02em',
+  lineHeight: 1.4,
+  transition: 'all 125ms ease-out',
+  textDecoration: 'none',
+  px: '$4',
+  py: '$2',
+  whiteSpace: 'nowrap',
+  width: 'max-content',
+  '&[disabled]': {
+    opacity: 0.35,
+    cursor: 'not-allowed'
+  },
+  variants: {
+    theme: {
+      primary: getButtonSolidVariant('$primary500', '$primary900'),
+      secondary: getButtonSolidVariant('$secondary500', '$secondary700'),
+      tertiary: getButtonSolidVariant('$tertiary500', '$tertiary700'),
+      success: getButtonSolidVariant('$success', '$successDark'),
+      warning: getButtonSolidVariant('$warning', '$warningDark'),
+      danger: getButtonSolidVariant('$danger', '$dangerDark')
+    },
+    variant: {
+      outline: {
+        boxShadow: 'inset 0 0 0 2px',
+        backgroundColor: 'white',
+        '&:not([disabled]):hover, &:not([disabled):focus': {
+          textDecoration: 'none',
+          backgroundColor: 'white'
+        },
+        '&[disabled]': { backgroundColor: 'white' }
+      }
+    }
+  }
+})
+
+StyledButton.compoundVariant(
+  {
+    theme: 'primary',
+    variant: 'outline'
+  },
+  getButtonOutlineVariant('$primary500', '$primary900')
+)
+
+StyledButton.compoundVariant(
+  {
+    theme: 'secondary',
+    variant: 'outline'
+  },
+  getButtonOutlineVariant('$secondary500', '$secondary900')
+)
+
+StyledButton.compoundVariant(
+  {
+    theme: 'tertiary',
+    variant: 'outline'
+  },
+  getButtonOutlineVariant('$tertiary500', '$tertiary700')
+)
+
+type ButtonProps = StitchesProps<typeof StyledButton>
+
+export const Button = ({
+  theme = 'primary',
```

Rather than setting the default value here, could we do the following?

```tsx
Button.defaultProps = {
  theme: 'primary'
} as Partial<ButtonProps>
```

There seems to be a bug in TS where default prop values assigned inline aren't recognised. This probably wouldn't cause a problem in the `Button` component, but it causes build errors for components with required props. Since those components (e.g. `Input`, which has a required `type` prop defaulted to `'text'`) need to use the long form, I think it would be good to use it everywhere so that we're consistent and if a new contributor looks at how we've done it before, they'll find a pattern that definitely works.

The `as Partial<ButtonProps>` casting ensures that we don't default to an invalid prop.

Here's an issue in the TS repo about this problem: https://github.com/microsoft/TypeScript/issues/31247

---

Yes Tom mentioned this yesterday when we talked. will do